src/h3/mod.rs

```diff
@@ -954,6 +995,54 @@ impl Connection {
         Err(Error::Done)
     }
 
+    /// Sends a GOAWAY frame to support graceful connection closure.
+    ///
+    /// When quiche is used in the server role, the `id` parameter is the stream
+    /// ID of the highest processed request. This can be any valid ID between 0
+    /// and 2^62-4. However, the ID cannot be increased. Failure to satisfy
+    /// these conditions will return an error.
+    ///
+    /// This method does not close the QUIC connection. Applications are
+    /// required to call [`close`] themselves.
+    ///
+    /// [`close()`]: ../struct.Connection.html#method.close
+    pub fn send_goaway(
+        &mut self, conn: &mut super::Connection, id: u64,
+    ) -> Result<()> {
+        if self.is_server && id % 4 != 0 {
```

Comment:
Since we don't support push, should we prevent quiche clients from calling this?

Reply:
I added an early exit and a TODO.